ui/src/utils/ui.ts

```diff
@@ -390,3 +390,12 @@ export function getElementCoordinates(element: HTMLElement) {
     clientHeight: element.clientHeight,
   };
 }
+
+/**
+ * TODO(andrewhead): Handle the case of arXiv publications that have multiple versions. How do we
+ * make sure we're querying for the same version of paper data as the paper that was opened?
+ */
+ export function extractArxivId(url: string): string | undefined {
+  const matches = url.match(/arxiv\.org\/pdf\/(.*)(?:\.pdf).*$/) || [];
+  return matches[1];
```

Comment:
(This is a question for me, not about your code)
Is your code prefering `undefined` or `null` for these situations?

Seems like `null` is the right option, because it means you have a result which has no value, while `undefined` means you don't even have a result (as in, you didn't return anything). I know TS can have configs for saying a type can have `null` for a value, or to require it to be declared explicitly (as you have done with `undefined`).

Reply:
I believe I authored the first version of this code. Given your explanation, I agree that, were I to write this again, I would have it return `null` if no arXiv ID was present.

Reply:
There's a fair bit of inconsistency in the code between returning `undefined` or `null` in places, but it would be good to consolidate on one. I'd also go `null`